servers/src/main/java/tachyon/web/WebInterfaceBrowseServlet.java

```diff
@@ -209,8 +213,14 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response)
       UiFileInfo toAdd = new UiFileInfo(fileInfo);
       try {
         if (!toAdd.getIsDirectory() && fileInfo.getLength() > 0) {
-          toAdd.setFileLocations(mMaster.getFileSystemMaster().getFileBlockInfoList(toAdd.getId())
-              .get(0).getLocations());
+          FileBlockInfo blockInfo =
+              mMaster.getFileSystemMaster().getFileBlockInfoList(toAdd.getId()).get(0);
+          List<NetAddress> addrs = Lists.newArrayList(blockInfo.getUnderFsLocations());
+          // add the in-memory block locations
+          for (BlockLocation location : blockInfo.getBlockInfo().getLocations()) {
```

Comment:
I think it would be nice to list the tachyon locations before the ufs locations.